### ISSUE TITLE
Scala module 2.10.0 requires Jackson Databind version >= 2.10.0 and <2.11.0

### ISSUE THREAD

**Issue by psarka** [lensesio/stream-reactor#637]
I'm trying to use the influxdb sink v.1.2.3-2.1.0 with the v.2.3 kafka brokers, which fails with the crash pasted bellow. Your readme states, that only kafka 2.1 is supported, so I'm wondering if this crash is caused by the unsupported version of kafka brokers.

If so, I would love to know if this can be somehow resolved, or if I should look for other solutions.

Thanks!

```
 textPayload: "org.apache.kafka.connect.errors.ConnectException: Exiting WorkerSinkTask due to unrecoverable exception.
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:561)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:322)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:224)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:192)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:177)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:227)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.ExceptionInInitializerError
	at com.datamountaineer.streamreactor.connect.influx.converters.SinkRecordParser$$anonfun$5.apply(SinkRecordParser.scala:59)
	at com.datamountaineer.streamreactor.connect.influx.converters.SinkRecordParser$$anonfun$5.apply(SinkRecordParser.scala:59)
	at scala.util.Try$.apply(Try.scala:192)
	at com.datamountaineer.streamreactor.connect.influx.converters.SinkRecordParser$.build(SinkRecordParser.scala:59)
	at com.datamountaineer.streamreactor.connect.influx.writers.InfluxBatchPointsBuilder$$anonfun$build$1.apply(InfluxBatchPointsBuilder.scala:90)
	at com.datamountaineer.streamreactor.connect.influx.writers.InfluxBatchPointsBuilder$$anonfun$build$1.apply(InfluxBatchPointsBuilder.scala:88)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
	at com.datamountaineer.streamreactor.connect.influx.writers.InfluxBatchPointsBuilder.build(InfluxBatchPointsBuilder.scala:88)
	at com.datamountaineer.streamreactor.connect.influx.writers.InfluxDbWriter.write(InfluxDbWriter.scala:45)
	at com.datamountaineer.streamreactor.connect.influx.InfluxSinkTask$$anonfun$put$2.apply(InfluxSinkTask.scala:77)
	at com.datamountaineer.streamreactor.connect.influx.InfluxSinkTask$$anonfun$put$2.apply(InfluxSinkTask.scala:77)
	at scala.Option.foreach(Option.scala:257)
	at com.datamountaineer.streamreactor.connect.influx.InfluxSinkTask.put(InfluxSinkTask.scala:77)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:539)
	... 10 more
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Scala module 2.10.0 requires Jackson Databind version >= 2.10.0 and < 2.11.0
	at com.fasterxml.jackson.module.scala.JacksonModule$class.setupModule(JacksonModule.scala:61)
	at com.fasterxml.jackson.module.scala.DefaultScalaModule.setupModule(DefaultScalaModule.scala:18)
	at com.fasterxml.jackson.databind.ObjectMapper.registerModule(ObjectMapper.java:751)
	at com.landoop.json.sql.JacksonJson$.<init>(JacksonJson.scala:26)
	at com.landoop.json.sql.JacksonJson$.<clinit>(JacksonJson.scala)
	... 31 more
```

**Comment by andrewstevenson** [comment]
@psarka How and where are you installing the connector?

**Comment by psarka** [comment]
I'm dropping the `kafka-connect-influxdb-1.2.3-2.1.0-all.jar` into the plugins folder of the kafka connect service, run by the `confluentinc/cp-kafka-connect:5.3.2` docker image.

**Comment by savulchik** [comment]
Also getting this exception in cassandra sink connector 1.2.0 after upgrading kafka connect to 2.4.0 that uses the following jackson jars
```
/opt/kafka_2.12-2.4.0/libs/jackson-module-scala_2.12-2.10.0.jar
/opt/kafka_2.12-2.4.0/libs/jackson-datatype-jdk8-2.10.0.jar
/opt/kafka_2.12-2.4.0/libs/jackson-core-2.10.0.jar
/opt/kafka_2.12-2.4.0/libs/jackson-jaxrs-json-provider-2.10.0.jar
/opt/kafka_2.12-2.4.0/libs/jackson-module-jaxb-annotations-2.10.0.jar
/opt/kafka_2.12-2.4.0/libs/jackson-jaxrs-base-2.10.0.jar
/opt/kafka_2.12-2.4.0/libs/jackson-dataformat-csv-2.10.0.jar
/opt/kafka_2.12-2.4.0/libs/jackson-annotations-2.10.0.jar
/opt/kafka_2.12-2.4.0/libs/jackson-databind-2.10.0.jar
/opt/kafka_2.12-2.4.0/libs/jackson-module-paranamer-2.10.0.jar
/opt/kafka_2.12-2.4.0/libs/jackson-core-2.9.5.jar
/opt/kafka_2.12-2.4.0/libs/jackson-databind-2.9.5.jar
/opt/kafka_2.12-2.4.0/libs/jackson-annotations-2.9.0.jar
```
and connector uses
```
/opt/connectors/kafka-connect-cassandra/jackson-core-asl-1.9.13.jar
/opt/connectors/kafka-connect-cassandra/jackson-jaxrs-base-2.9.6.jar
/opt/connectors/kafka-connect-cassandra/jackson-module-jaxb-annotations-2.9.6.jar
/opt/connectors/kafka-connect-cassandra/jackson-jaxrs-json-provider-2.9.6.jar
/opt/connectors/kafka-connect-cassandra/jackson-core-2.9.6.jar
/opt/connectors/kafka-connect-cassandra/jackson-module-scala_2.11-2.9.1.jar
/opt/connectors/kafka-connect-cassandra/jackson-mapper-asl-1.9.13.jar
/opt/connectors/kafka-connect-cassandra/jackson-module-paranamer-2.9.1.jar
/opt/connectors/kafka-connect-cassandra/jackson-databind-2.9.6.jar
/opt/connectors/kafka-connect-cassandra/jackson-annotations-2.9.6.jar
``` 

The complete exceptions stack trace is
```
c.f.j.d.JsonMappingException: Scala module 2.10.0 requires Jackson Databind version >= 2.10.0 and < 2.11.0
	at c.f.j.m.s.JacksonModule$class.setupModule(JacksonModule.scala:61)
	at c.f.j.m.s.DefaultScalaModule.setupModule(DefaultScalaModule.scala:18)
	at c.f.j.d.ObjectMapper.registerModule(ObjectMapper.java:751)
	at c.l.j.s.JacksonJson$.<init>(JacksonJson.scala:26)
	at c.l.j.s.JacksonJson$.<clinit>(JacksonJson.scala)
	... 18 common frames omitted
Wrapped by: j.l.ExceptionInInitializerError: null
	at c.d.s.c.c.Transform$.apply(Transform.scala:93)
	at c.d.s.c.c.s.CassandraJsonWriter$$anonfun$com$datamountaineer$streamreactor$connect$cassandra$sink$CassandraJsonWriter$$insert$1.apply(CassandraJsonWriter.scala:182)
	at c.d.s.c.c.s.CassandraJsonWriter$$anonfun$com$datamountaineer$streamreactor$connect$cassandra$sink$CassandraJsonWriter$$insert$1.apply(CassandraJsonWriter.scala:181)
	at s.c.i.Map$Map1.foreach(Map.scala:116)
	at c.d.s.c.c.s.CassandraJsonWriter.com$datamountaineer$streamreactor$connect$cassandra$sink$CassandraJsonWriter$$insert(CassandraJsonWriter.scala:181)
	at c.d.s.c.c.s.CassandraJsonWriter$$anonfun$4$$anonfun$apply$1$$anonfun$apply$mcV$sp$1.apply(CassandraJsonWriter.scala:161)
	at c.d.s.c.c.s.CassandraJsonWriter$$anonfun$4$$anonfun$apply$1$$anonfun$apply$mcV$sp$1.apply(CassandraJsonWriter.scala:159)
	at s.c.Iterator$class.foreach(Iterator.scala:891)
	at s.c.AbstractIterator.foreach(Iterator.scala:1334)
	at s.c.IterableLike$class.foreach(IterableLike.scala:72)
	at s.c.AbstractIterable.foreach(Iterable.scala:54)
	at c.d.s.c.c.s.CassandraJsonWriter$$anonfun$4$$anonfun$apply$1.apply$mcV$sp(CassandraJsonWriter.scala:159)
	at c.d.s.c.c.s.CassandraJsonWriter$$anonfun$4$$anonfun$apply$1.apply(CassandraJsonWriter.scala:159)
	at c.d.s.c.c.s.CassandraJsonWriter$$anonfun$4$$anonfun$apply$1.apply(CassandraJsonWriter.scala:159)
	at c.d.s.c.c.ExecutorExtension$RunnableWrapper$$anon$1.run(ExecutorExtension.scala:30)
	... 3 common frames omitted
Wrapped by: j.u.c.ExecutionException: Boxed Error
	at s.c.impl.Promise$.resolver(Promise.scala:59)
	at s.c.impl.Promise$.scala$concurrent$impl$Promise$$resolveTry(Promise.scala:51)
	at s.c.i.Promise$DefaultPromise.tryComplete(Promise.scala:248)
	at s.c.Promise$class.complete(Promise.scala:55)
	at s.c.i.Promise$DefaultPromise.complete(Promise.scala:157)
	at s.c.Promise$class.failure(Promise.scala:104)
	at s.c.i.Promise$DefaultPromise.failure(Promise.scala:157)
	at c.d.s.c.c.ExecutorExtension$RunnableWrapper$$anon$1.run(ExecutorExtension.scala:29)
	... 3 common frames omitted
Wrapped by: o.a.k.c.e.RetriableException: java.util.concurrent.ExecutionException: Boxed Error
	at c.d.s.c.e.RetryErrorPolicy.handle(ErrorPolicy.scala:70)
	at c.d.s.c.e.ErrorHandler$class.handleError(ErrorHandler.scala:83)
	at c.d.s.c.e.ErrorHandler$class.handleTry(ErrorHandler.scala:64)
	at c.d.s.c.c.s.CassandraJsonWriter.handleTry(CassandraJsonWriter.scala:46)
	at c.d.s.c.c.s.CassandraJsonWriter.write(CassandraJsonWriter.scala:175)
	at c.d.s.c.c.s.CassandraJsonWriter.write(CassandraJsonWriter.scala:139)
	at c.d.s.c.c.s.CassandraSinkTask$$anonfun$put$2.apply(CassandraSinkTask.scala:69)
	at c.d.s.c.c.s.CassandraSinkTask$$anonfun$put$2.apply(CassandraSinkTask.scala:69)
	at scala.Option.foreach(Option.scala:257)
	at c.d.s.c.c.s.CassandraSinkTask.put(CassandraSinkTask.scala:69)
	at o.a.k.c.r.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:539)
	at o.a.k.c.r.WorkerSinkTask.poll(WorkerSinkTask.java:322)
	at o.a.k.c.r.WorkerSinkTask.iteration(WorkerSinkTask.java:224)
	at o.a.k.c.r.WorkerSinkTask.execute(WorkerSinkTask.java:192)
	at o.a.k.c.r.WorkerTask.doRun(WorkerTask.java:177)
	at o.a.k.c.r.WorkerTask.run(WorkerTask.java:227)
	at j.u.c.Executors$RunnableAdapter.call(Executors.java:511)
	at j.u.c.FutureTask.run(FutureTask.java:266)
	at j.u.c.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at j.u.c.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by savulchik** [comment]
Also happens with kafka connect 2.3.1 but not with 2.3.0
```
/opt/kafka_2.12-2.3.1/libs/jackson-module-scala_2.12-2.10.0.jar
/opt/kafka_2.12-2.3.1/libs/jackson-datatype-jdk8-2.10.0.jar
/opt/kafka_2.12-2.3.1/libs/jackson-core-2.10.0.jar
/opt/kafka_2.12-2.3.1/libs/jackson-jaxrs-json-provider-2.10.0.jar
/opt/kafka_2.12-2.3.1/libs/jackson-module-jaxb-annotations-2.10.0.jar
/opt/kafka_2.12-2.3.1/libs/jackson-jaxrs-base-2.10.0.jar
/opt/kafka_2.12-2.3.1/libs/jackson-dataformat-csv-2.10.0.jar
/opt/kafka_2.12-2.3.1/libs/jackson-annotations-2.10.0.jar
/opt/kafka_2.12-2.3.1/libs/jackson-databind-2.10.0.jar
/opt/kafka_2.12-2.3.1/libs/jackson-module-paranamer-2.10.0.jar
/opt/kafka_2.12-2.3.1/libs/jackson-core-2.9.5.jar
/opt/kafka_2.12-2.3.1/libs/jackson-databind-2.9.5.jar
/opt/kafka_2.12-2.3.1/libs/jackson-annotations-2.9.0.jar
```
```
/opt/kafka_2.12-2.3.0/libs/jackson-jaxrs-json-provider-2.9.9.jar
/opt/kafka_2.12-2.3.0/libs/jackson-module-paranamer-2.9.9.jar
/opt/kafka_2.12-2.3.0/libs/jackson-databind-2.9.9.jar
/opt/kafka_2.12-2.3.0/libs/jackson-core-2.9.9.jar
/opt/kafka_2.12-2.3.0/libs/jackson-annotations-2.9.9.jar
/opt/kafka_2.12-2.3.0/libs/jackson-module-jaxb-annotations-2.9.9.jar
/opt/kafka_2.12-2.3.0/libs/jackson-module-scala_2.12-2.9.9.jar
/opt/kafka_2.12-2.3.0/libs/jackson-jaxrs-base-2.9.9.jar
/opt/kafka_2.12-2.3.0/libs/jackson-datatype-jdk8-2.9.9.jar
/opt/kafka_2.12-2.3.0/libs/jackson-dataformat-csv-2.9.9.jar
/opt/kafka_2.12-2.3.0/libs/jackson-core-2.9.5.jar
/opt/kafka_2.12-2.3.0/libs/jackson-databind-2.9.5.jar
/opt/kafka_2.12-2.3.0/libs/jackson-annotations-2.9.0.jar
```

**Comment by zzztimbo** [comment]
I too am having the same issue. I'm using strimzi. 

```
FROM strimzi/kafka:0.16.1-kafka-2.4.0
USER root:root
RUN mkdir -p /opt/kafka/plugins/stream-reactor
COPY ./kafka-connect-cassandra-1.2.7-2.1.0-all.jar /opt/kafka/plugins/stream-reactor/
USER 1001
```

**Comment by andrewstevenson** [comment]
@zzztimbo We are preparing a release with scala 2.12 that will address this.

**Comment by andrewstevenson** [comment]
Fixed in #669
